include/knowhere/index_node.h

```diff
@@ -115,6 +115,72 @@ class IndexNode : public Object {
     Version version_;
 };
 
+// An iterator implementation that accepts a list of distances and ids and returns them in order.
+class PrecomputedDistanceIterator : public IndexNode::iterator {
+ public:
+    PrecomputedDistanceIterator(std::vector<std::pair<float, int64_t>>&& distances_ids, bool larger_is_closer)
+        : comp_(larger_is_closer), results_(std::move(distances_ids)) {
+        sort_size_ = std::max((size_t)50000, results_.size() / 10);
+        sort_next();
+    }
+
+    // Construct an iterator from a list of distances with index being id, filtering out zero distances.
+    PrecomputedDistanceIterator(std::vector<float> distances, bool larger_is_closer) : comp_(larger_is_closer) {
+        for (size_t i = 0; i < distances.size(); i++) {
```

Comment:
you could add `results_.reserve(distances.size());`, if it is expected that the number of zeros among `distances` is small

Reply:
I used `results_.reserve(distances.size() * 0.3);`. 30% is a ratio guesstimate of non-zero distances: probability of 2 random sparse splade vectors(100 non zero dims out of 30000 total dims) sharing at least 1 common non-zero dimension.